### ISSUE TITLE
The config table is missing a foreign key to tenant

### ISSUE THREAD

**Issue by thomashilzendegen** [thinktecture/relayserver#461]
We have to add this.

**Comment by fschuetz04** [comment]
Apparently, EF core automatically added the column `ConfigTenantName` to `Tenants` as a foreign key referencing `Configs.TenantName`. Obviously, this is not the correct relationship.

When describing the `Tenants` table in the PostgreSQL CLI, the following is shown:

```sql
                                    Table "public.Tenants"
               Column               |          Type           | Collation | Nullable | Default
------------------------------------+-------------------------+-----------+----------+---------
 NormalizedName                     | character varying(100)  |           | not null |
 Name                               | character varying(100)  |           | not null |
 DisplayName                        | character varying(200)  |           |          |
 Description                        | character varying(1000) |           |          |
 ConfigTenantName                   | text                    |           |          |
 RequireAuthentication              | boolean                 |           | not null | false
 MaximumConcurrentConnectorRequests | integer                 |           | not null | 0
Indexes:
    "PK_Tenants" PRIMARY KEY, btree ("NormalizedName")
    "IX_Tenants_ConfigTenantName" btree ("ConfigTenantName")
    "IX_Tenants_Name" UNIQUE, btree ("Name")
Foreign-key constraints:
    "FK_Tenants_Configs_ConfigTenantName" FOREIGN KEY ("ConfigTenantName") REFERENCES "Configs"("TenantName")
Referenced by:
    TABLE ""ClientSecrets"" CONSTRAINT "FK_ClientSecrets_Tenants_TenantName" FOREIGN KEY ("TenantName") REFERENCES "Tenants"("NormalizedName") ON DELETE CASCADE
    TABLE ""Connections"" CONSTRAINT "FK_Connections_Tenants_TenantName" FOREIGN KEY ("TenantName") REFERENCES "Tenants"("NormalizedName") ON DELETE CASCADE
    TABLE ""Requests"" CONSTRAINT "FK_Requests_Tenants_TenantName" FOREIGN KEY ("TenantName") REFERENCES "Tenants"("NormalizedName") ON DELETE CASCADE
```

**Comment by gingters** [comment]
Fixed by #470
